lib/matplotlib/tests/test_datetime.py

```diff
@@ -32,11 +32,27 @@ def test_axhline(self):
         fig, ax = plt.subplots()
         ax.axhline(...)
 
-    @pytest.mark.xfail(reason="Test for axhspan not written yet")
     @mpl.style.context("default")
     def test_axhspan(self):
-        fig, ax = plt.subplots()
-        ax.axhspan(...)
+        mpl.rcParams["date.converter"] = 'concise'
+        np.random.seed(19680801)
+
+        start_date = datetime.datetime(2023, 1, 1)
+        time_delta = datetime.timedelta(days=1)
+
+        fig, (ax1, ax2) = plt.subplots(2, 1, constrained_layout=True, figsize=(10, 8))
```

Comment:
The two subplots aren't really testing anything different right now. Could you please expand to 3 subplots and match the behavior I suggested here on your test_axvspan PR? https://github.com/matplotlib/matplotlib/pull/27130#discussion_r1365812782

Reply:
Hello, i made new tests and updated image in PR. Can you please check it? If it´s OK i will do the same for axvspan in my other PR.

Reply:
I think that looks much better, thanks!